src/main/java/com/mewna/catnip/shard/manager/DefaultShardManager.java

```diff
@@ -173,74 +169,64 @@ private void loadShards() {
-        final String deployment = shards.remove(id);
-        if(deployment != null) {
-            catnip().vertx().undeploy(deployment);
+        final CatnipShardImpl shard = shards.remove(id);
+        if(shard != null) {
+            shard.stop();
         }
     }
     
+    @SuppressWarnings("ResultOfMethodCallIgnored")
     private void connectShard(final int id) {
-        final String deploymentId = shards.get(id);
+        final CatnipShardImpl shard = shards.get(id);
         
-        if(deploymentId == null) {
+        if(shard == null) {
             catnip().logAdapter().error("Cannot find deployment ID of shard {}, re-queueing...", id);
             addToConnectQueue(id);
             return;
         }
         
-        catnip().eventBus().<ShardConnectState>send(computeAddress(CONTROL, id), CONNECT,
-                new DeliveryOptions().setSendTimeout(60_000), // allow longer timeouts because of network memes.
-                result -> {
-                    // ignore the reply if deployment ID differs.
-                    if(!deploymentId.equals(shards.get(id))) {
-                        return;
-                    }
-                    
-                    if(result.failed()) {
-                        catnip().logAdapter().error("Something went really wrong while trying to connect shard {}, re-queueing...", id, result.cause());
-                        addToConnectQueue(id);
-                        return;
-                    }
-                    
-                    final ShardConnectState state = result.result().body();
-                    switch(state) {
-                        case READY:
-                            catnip().logAdapter().info("Connected shard {}(/{})", id, shardCount);
-                            break;
-                        case RESUMED:
-                            catnip().logAdapter().info("Resumed shard {}(/{})", id, shardCount);
-                            break;
-                        case FAILED:
-                            catnip().logAdapter().error("Failed connecting shard {}(/{}), re-queueing...", id, shardCount);
-                            addToConnectQueue(id);
-                            break;
-                        case INVALID:
-                            catnip().logAdapter().error("Invalid session on shard {}(/{}), re-queueing...", id, shardCount);
-                            addToConnectQueue(id);
-                            break;
-                        case CANCEL:
-                            break; // do nothing
-                        default:
-                            catnip().logAdapter().error("This shouldn't happen, but we got unknown state for shard {}: {}", id, state);
-                            break;
-                    }
-                    
-                    conditions().forEach(e -> e.postshard(id, state));
-                });
+        shard.connect().timeout(1, TimeUnit.MINUTES).subscribe(state -> {
```

Comment:
1 minute seems like a pretty long timeout ngl

Reply:
yeah I just converted 60000 ms to... uh... 1 minute?